server/src/main/java/org/elasticsearch/script/VectorScoreScriptUtils.java

```diff
@@ -51,22 +55,54 @@ public static class ByteDenseVectorFunction extends DenseVectorFunction {
          * @param scoreScript The script in which this function was referenced.
          * @param field The vector field.
          * @param queryVector The query vector.
+         * @param normalizeFloatQuery {@code true} if the query vector is a float vector, then normalize it.
+         * @param allowedTypes The types the vector is allowed to be.
          */
-        public ByteDenseVectorFunction(ScoreScript scoreScript, DenseVectorDocValuesField field, List<Number> queryVector) {
+        public ByteDenseVectorFunction(
+            ScoreScript scoreScript,
+            DenseVectorDocValuesField field,
+            List<Number> queryVector,
+            boolean normalizeFloatQuery,
+            ElementType... allowedTypes
+        ) {
             super(scoreScript, field);
             field.getElementType().checkDimensions(field.get().getDims(), queryVector.size());
-            this.queryVector = new byte[queryVector.size()];
-            float[] validateValues = new float[queryVector.size()];
-            int queryMagnitude = 0;
+            float[] floatValues = new float[queryVector.size()];
             for (int i = 0; i < queryVector.size(); i++) {
-                final Number number = queryVector.get(i);
-                byte value = number.byteValue();
-                this.queryVector[i] = value;
-                queryMagnitude += value * value;
-                validateValues[i] = number.floatValue();
+                floatValues[i] = queryVector.get(i).floatValue();
```

Comment:
let's calculate the magnitude in this loop like you did before. This allows for simple normalization later for floating point, but we keep the magnitude around for `byte[]` since you cannot normalize `byte[]` values.

Reply:
Doh, yes, the magnitude is calculated for byte too...